rerun_py/rerun_sdk/rerun/blueprint/api.py

```diff
@@ -129,22 +130,23 @@ def _log_to_stream(self, stream: RecordingStream) -> None:
             visible=self.visible,
         )
 
-        stream.log(self.blueprint_path(), arch, recording=stream)  # type: ignore[attr-defined]
+        stream.log(self.blueprint_path(), arch)
 
         for prop_name, prop in self.properties.items():
-            stream.log(f"{self.blueprint_path()}/{prop_name}", prop, recording=stream)  # type: ignore[attr-defined]
+            stream.log(f"{self.blueprint_path()}/{prop_name}", prop)
 
         for default in self.defaults:
-            if hasattr(default, "as_component_batches"):
-                stream.log(f"{self.blueprint_path()}/defaults", default, recording=stream)  # type: ignore[attr-defined]
-            elif hasattr(default, "component_descriptor"):
-                stream.log(f"{self.blueprint_path()}/defaults", [default], recording=stream)  # type: ignore[attr-defined]
+            if isinstance(default, AsComponents):
+                stream.log(f"{self.blueprint_path()}/defaults", default)
+            elif isinstance(default, ComponentBatchLike):
+                stream.log(f"{self.blueprint_path()}/defaults", [default])  # type: ignore[list-item]
             else:
                 raise ValueError(f"Provided default: {default} is neither a component nor a component batch.")
 
         for path, components in self.overrides.items():
-            stream.log(  # type: ignore[attr-defined]
-                f"{self.blueprint_path()}/ViewContents/individual_overrides/{path}", components, recording=stream
+            stream.log(
+                f"{self.blueprint_path()}/ViewContents/individual_overrides/{path}",
+                components,  # type: ignore[arg-type]
             )
```

Comment:
Not sure what happened here, but I assume you know what you're doing

Reply:
Yes. This is a first step towards removing all these `# type: ignore`, but there is more work to do.